test/test_instance_requirements.rb

```diff
@@ -374,6 +377,22 @@
             instanciated = req.instanciate(plan)
             assert_equal srv, instanciated.model
         end
+
+        it "accepts a service selection expressed as an InstanceRequirements" do
+            srv_m  = Syskit::DataService.new_submodel
+            task_m = Syskit::TaskContext.new_submodel do
+                argument :key
+                provides srv_m, as: "test"
+            end
+
+            req = Syskit::InstanceRequirements.new([task_m])
+            srv = task_m.with_arguments(key: 10).test_srv
+            srv.name = "name" # closer to the original bug, which was a Definition
```

Comment:
what was the original bug? (is this a bugfix?)

Reply:
Well ... it can be also seen as "something unsupported is now supported" ...

Was injecting in a profile something that looks exactly like `task_m.with_arguments(key: 10).test_srv` and it started to fail.